propagate errors from userscript master task

todfreed opened the pull request:
Makes it possible for MoonGen to exit nonzero when the userscript fails

emmericp commented:
good idea, but the code looks broken syntactically; also, it would favor calling os.exit() explicitly here to get the return code, the second error reduces the readability

todfreed commented:
- fixed the syntax error
- changed to os.exit instead of error

The backtrace is already printed, since xpcall is invoked with getStackTrace.

I agree it's better without having two exceptions to resolve.

emmericp commented:
thank you!
